client/context/query.go

```diff
@@ -72,40 +78,62 @@ func (ctx CLIContext) GetFromName() string {
 	return ctx.FromName
 }
 
-// query performs a query to a Tendermint node with the provided store name
-// and path. It returns the result and height of the query upon success
-// or an error if the query fails. In addition, it will verify the returned
-// proof if TrustNode is disabled. If proof verification fails or the query
-// height is invalid, an error will be returned.
-func (ctx CLIContext) query(path string, key tmbytes.HexBytes) (res []byte, height int64, err error) {
+func (ctx CLIContext) queryABCI(req abci.RequestQuery) (abci.ResponseQuery, error) {
 	node, err := ctx.GetNode()
 	if err != nil {
-		return res, height, err
+		return abci.ResponseQuery{}, err
+	}
+
+	// When a client did not provide a query height, manually query for it so it can
```

Comment:
This is not needed. The BaseApp already handles this. In fact, I had this originally the `CLIContext` querying logic but then removed it because I recall it didn't work properly and the BaseApp is the place where it should be done.